### ISSUE TITLE
[Syncfusion Flutter Charts]Crash when plotting the BollingerBand indicator

### ISSUE THREAD

**Issue by khanh-nguyen-ngoc** [syncfusion/flutter-widgets#844]
I have a dataset like the one attached below. When plotting the candle chart, it's totally fine. However, when I add the Bollinger band indicator for this dataset, the chart crashes with this log.
[1324_fake.txt](https://github.com/syncfusion/flutter-widgets/files/9278204/1324_fake.txt)

```
======== Exception caught by widgets library =======================================================
The following UnsupportedError was thrown building LayoutBuilder:
Unsupported operation: Infinity or NaN toInt

The relevant error-causing widget was: 
  SfCartesianChart SfCartesianChart:file:///Users/teq-macm1mobile-5/Desktop/projects/stream-flutter/lib/features/home/stocks/stock_details/chart_details/stock_details_chart_widget.dart:127:45
When the exception was thrown, this was the stack: 
#0      double.floor (dart:core-patch/double.dart)
#1      ChartAxisRendererDetails.calculateNumericNiceInterval (package:syncfusion_flutter_charts/src/chart/axis/axis.dart:3831:64)
#2      NumericAxisDetails._calculateActualRange (package:syncfusion_flutter_charts/src/chart/axis/numeric_axis.dart:759:9)
#3      NumericAxisDetails.calculateRangeAndInterval (package:syncfusion_flutter_charts/src/chart/axis/numeric_axis.dart:713:5)
#4      ChartAxisPanel.measureAxesBounds (package:syncfusion_flutter_charts/src/chart/axis/axis_panel.dart:214:21)
#5      ContainerArea._calculateBounds (package:syncfusion_flutter_charts/src/chart/base/chart_base.dart:2565:32)
#6      ContainerArea._initializeChart (package:syncfusion_flutter_charts/src/chart/base/chart_base.dart:2548:5)
#7      ContainerArea.build.<anonymous closure> (package:syncfusion_flutter_charts/src/chart/base/chart_base.dart:2539:34)
#8      _LayoutBuilderElement._layout.layoutCallback (package:flutter/src/widgets/layout_builder.dart:119:77)
#9      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2597:19)
#10     _LayoutBuilderElement._layout (package:flutter/src/widgets/layout_builder.dart:153:12)
#11     RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:2025:59)
#12     PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:918:15)
#13     RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:2025:14)
#14     RenderConstrainedLayoutBuilder.rebuildIfNecessary (package:flutter/src/widgets/layout_builder.dart:227:7)
#15     _RenderLayoutBuilder.performLayout (package:flutter/src/widgets/layout_builder.dart:316:5)
#16     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1757:7)
#17     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:887:18)
#18     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:504:19)
#19     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:892:13)
#20     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:370:5)
#21     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1146:15)
#22     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1083:9)
#23     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:997:5)
#27     _invoke (dart:ui/hooks.dart:151:10)
#28     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:308:5)
#29     _drawFrame (dart:ui/hooks.dart:115:31)
(elided 3 frames from dart:async)
====================================================================================================
```

**Comment by Yuvaraj-Gajaraj** [comment]
Hi @khanh-nguyen-ngoc,

We have checked the reported issue with [CandleSeries](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/CandleSeries-class.html) and the [BollingerBandIndicator](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/BollingerBandIndicator-class.html) based on the shared data source in the query. But the reported issue is not reproduced at our end. We have shared the sample below for your reference. Please modify the sample as per your requirement to replicate the issue, which will help us to assist you in a better way.

Sample: 
[i397054.zip](https://github.com/syncfusion/flutter-widgets/files/9296835/i397054.zip)


Regards,
Yuvaraj.

**Comment by SriramKiranSenthilkumar** [comment]
Hi @khanh-nguyen-ngoc,

Sorry for the delay and our development team is checking the reported issue at our end. We will update further details in one business day, and we appreciate your patience until then.

Regards,
Sriram Kiran

**Comment by SriramKiranSenthilkumar** [comment]
Hi @khanh-nguyen-ngoc,

We can reproduce the reported issue at our end, and we are analyzing the standard deviation calculation for Bollinger band algorithm in the source level for behavior confirmation. We will update further details in four business days, and we appreciate your patience until then.

Regards,
Sriram Kiran

**Comment by khanh-nguyen-ngoc** [comment]
> Hi @khanh-nguyen-ngoc,
> 
> We can reproduce the reported issue at our end, and we are analyzing the standard deviation calculation for Bollinger band algorithm in the source level for behavior confirmation. We will update further details in four business days, and we appreciate your patience until then.
> 
> Regards, Sriram Kiran

Thank you and your team for the effort. I'll go with my own implementation for now while waiting for your fix. If you guys need any support, just reply to this comment.

**Comment by SriramKiranSenthilkumar** [comment]
Hi @khanh-nguyen-ngoc,

Sorry for the delay. We have validated your queries at our end and for the nan exception issue which occurs when providing same number of close values alone for the Bollinger band, we have considered this as a bug and logged bug report for it. We will include the fix for this in our upcoming weekly patch release which will be rolled out by 1st week of October.

Regarding the Standard deviation miscalculation, we have checked the standard deviation calculation in our Bollinger band algorithm and logically the results were correct in our end. So, we kindly request you to share us your implementation logic along with data and output screenshot if possible so that it will be helpful in further validation at our end to assist you in a better way.

Regards,
Sriram Kiran

**Comment by khanh-nguyen-ngoc** [comment]
> Hi @khanh-nguyen-ngoc,
> 
> Sorry for the delay. We have validated your queries at our end and for the nan exception issue which occurs when providing same number of close values alone for the Bollinger band, we have considered this as a bug and logged bug report for it. We will include the fix for this in our upcoming weekly patch release which will be rolled out by 1st week of October.
> 
> Regarding the Standard deviation miscalculation, we have checked the standard deviation calculation in our Bollinger band algorithm and logically the results were correct in our end. So, we kindly request you to share us your implementation logic along with data and output screenshot if possible so that it will be helpful in further validation at our end to assist you in a better way.
> 
> Regards, Sriram Kiran

Hi @SriramKiranSenthilkumar,

I have set up a demo for my implementation of the Bollinger band calculation algorithm in the attached [main.txt](https://github.com/syncfusion/flutter-widgets/files/9429957/main.txt) file. the data I prepared in this demo is obtained from [this link](https://www.wsj.com/market-data/quotes/futures/ES00/advanced-chart) with the Bollinger band config is multiplier == 2 and period == 25. As you can see from the screenshot, my calculation has a very minor difference compared to the site's original bollinger band data.
<img width="728" alt="image" src="https://user-images.githubusercontent.com/85327714/186825891-32a887af-876e-4b82-8da6-55001641e503.png">


In summary, the main difference between my implementation and yours is the standard deviation calculation for each point

**Comment by SriramKiranSenthilkumar** [comment]
Hi @khanh-nguyen-ngoc,

We will validate your implementation and will update the further details after the analyzation is completed at our end.

Regards,
Sriram Kiran

**Comment by natrayansf** [comment]
Hi khanh-nguyen-ngoc,

We would like to let you know that the reported issue ‘Infinity or NaN toInt exception thrown when rendering Bollinger band with mapping more number identical close’ was already fixed and rolled out in our weekly patch release on 18th October 2022. So, we kindly request you to upgrade the SfCartesianChart package to the latest version below.

Version link – https://pub.dev/packages/syncfusion_flutter_charts/versions/20.3.50

Regards,
Natrayan
